source/Mlos.Python/mlos/Spaces/HypergridAdapters/unit_tests/TestCategoricalToOneHotEncodedHypergridAdapter.py

```diff
@@ -0,0 +1,173 @@
+        adaptee = self.simple_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True)
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    def test_projecting_dataframe_from_flat_to_one_hot_encoded_cross_product_drop_first_hypergrid(self):
+        adaptee = self.simple_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True, drop='first')
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    # Hierarchical Hypergrid tests
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee)
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_drop_first_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, drop='first')
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_cross_product_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True)
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    def test_projecting_dataframe_from_hierarchical_to_one_hot_encoding_cross_product_drop_first_hypergrid(self):
+        adaptee = self.hierarchical_hypergrid
+        adapter = CategoricalToOneHotEncodedHypergridAdapter(adaptee=adaptee, merge_all_categorical_dimensions=True, drop='first')
+        self._test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(adapter=adapter, adaptee=adaptee, num_random_points=1000)
+
+    # Helper functions
+    def _test_projecting_dataframe_categorical_to_one_hot_encoding_point_from_adaptee(self, adaptee, adapter, num_random_points: int):
+        original_df = adaptee.random_dataframe(num_samples=num_random_points)
+        projected_df = adapter.project_dataframe(df=original_df, in_place=False)
+        self.assertTrue(id(original_df) != id(projected_df))
+        for column in adapter.get_one_hot_encoded_column_names():
+            self.assertTrue(projected_df[column].between(0, 1).all())

```

Comment:
Would it be good to check if `projected_df[column].isin([0,1]).all()` instead of `between(0, 1)`?